### ISSUE TITLE
Validate batch checksum

### ISSUE THREAD

**Issue by Nevon** [tulios/kafkajs#17]
A lot of the errors we've seen have been because we've been reading the wrong amount of bytes for each message or batch. All of those errors would have been caught if we were validating the checksum that we get with each batch/message.

As it could imply a performance hit, we might want to conditionally use that validation.

----

There's a pretty good reference on how to perform the checksum validation on the batch level in the Java client:
https://github.com/apache/kafka/blob/647afeff6a2e3fd78328f6989e8d9f96bcde5121/clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java#L37-L76

On the message level, things look more dire:

https://github.com/apache/kafka/blob/cea319a4ad9c55d3d3263cf7a4224c25772d0e11/clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerRecord.java#L197-L204

For v2 and above, there is no checksum, because the checksum that was computed by the producer might not match what the broker sends, intentionally.

So basically we could do verification on each batch, and throw a retriable error if the checksum doesn't match.

**Comment by tulios** [comment]
If this appears again we can reconsider, closing for now
